docs/design/scheduler-simulate-design.md

```diff
@@ -0,0 +1,32 @@
+# Volcano scheduler simulate
+## background
+* Consider such situation: users changed the parameter of nodeorder plugin and need to know the effect to the production enviroment. For example, after change the mostrequested.weight, if the average wait time of big task is shorter than before, etc.
+* Currently, users must use real cluster to verify the effect, which is very time consuming and need a lot of node resources.
+* Scheduler simulate is an effect solution of the problem above. It runs fast and needs much lesser resources than real cluster.
+## design
+* The simulator includes simulator command, simulate-data-input, simulate-result-output, time simulator, kube-apiserver simulator
+![scheduler-simulate](images/scheduler-simulate.png) 
+### simulator command
+* scheduler-simulate can be runned by command: vc-scheduler-simulate --running_time=24h --config_path=./config --output_path=./result
+### simulate-data-input
+* Simulate data includes nodes.csv, pods.csv, queues.csv. These files can be put in config_path directory.
+* nodes.csv includes cpu_allocatable, memory_allocatable, label, maxPodNum
+* pods.csv includes cpu_request, memory_request, runsec, cron, createtime, nodeSelector, priority, queueName
+* queues.csv includes queueName, quota
+#### pod
+#### queue
+### simulate-result-output
+#### file
+* output file includes nodes_detail.csv, queues_detail.csv, pods_detail.csv, pods_detail.csv. These files will be put in output_path directory.
+* nodes_detail.csv includes ts, nodeName, cpuRequest, memoryRequest
+* queues_detail.csv includes ts, queueName, cpuRequest, memoryRequest
+* pods_detail.csv includes podName, createTs, scheduleTs, finishTs
+#### metrics
+* output metrics includes volcano_simulator_scheduler_wait_time, volcano_simulator_pod_run_time, volcano_simulator_node_cpu_request, volcano_simulator_node_memory_request which will help user to monitor the schduler simulate by prometheus.
+### time simulator
+* Time simulator is helpful to shorten simulate time because it will not get the time of real world, it will always get next timestamp of the min value between the create time of next pod and the finish time of next pod.
+* The time related parameter should get from time simulator like pod create time, pod finish time, current time, etc...
+### kube-apiserver simulator
```

Comment:
What about using kwok? kwok can simulate thousands of nodes and doesn't consume many resources. I'm wondering that if we use kwok, then we don't need to simulate kube-apiserver and kube-controller-manager. We can keep using kube-apiserver's and kube-controller-manager's features and ensure their reliability (if they meet some errors and failed ). I worry about if you need to simulate kube-apiserver and kcm, there is lots of work you need to do, and you can't ensure the reliability of kube-apiserver and kcm.

Reply:
The project kube-scheduler-simulator also follow kwok to simulate scheduling, can we also do this way? https://github.com/kubernetes-sigs/kube-scheduler-simulator

Reply:
The problem is how to speed up the simulation time, for example, a pod may run 20 hours and we also let it run 20 hours in kwok?

Reply:
I create a PR: #3830 to add two scripts: one is for installing kwok and the other is for creating fake nodes. Perhaps building upon kwok is a better way to do simulating, you can add your stage confugration to do time simulating or other useful simulating. I think it's better and help us to do less work, kwok has been adopted by many schedulers as a tool for simulating scheduling.

Reply:
I added kwok to the proposal to replace apiserver-simulator @JesseStutler